openfe/protocols/openmm_rfe/equil_rfe_methods.py

```diff
@@ -62,6 +62,73 @@ def _get_resname(off_mol) -> str:
     return names[0]
 
 
+def _validate_alchemical_components(
+        alchemical_components: dict[str, list[Component]],
+        mapping: Optional[dict[str, ComponentMapping]],
```

Comment:
since this function is very specific, no reason to have mapping be `Optional` here

Reply:
It's a chicken and egg thing right? mapping in protocols are `Optional` by default, so we need to have a place to catch `None` and throw an error... which is this method 😅